CMakeLists.txt

```diff
@@ -1,17 +1,84 @@
 cmake_minimum_required(VERSION 3.10)
 
-project(cxxeffect)
+project(cxxeffect CXX)
 
-# add the executable
-add_executable(cxxeffect
-    src/stream.hpp
-    src/stream.cpp
-    src/task.hpp
-    src/task.cpp
-    src/eff.hpp
-    src/eff.cpp)
+# We want to export compile commands b/c it's useful for ccls
+set(CMAKE_EXPORT_COMPILE_COMMANDS true)
 
-set_property(TARGET cxxeffect PROPERTY CXX_STANDARD 17)
+# cxxeffect is an interface library b/c it has a lot of templated code
+# That means it's best implemented as header-only without the need for cpp files
+add_library(cxxeffect INTERFACE)
 
-target_link_libraries(cxxeffect pthread)
+set_property(TARGET cxxeffect PROPERTY CXX_STANDARD_REQUIRED 20)
 
+target_compile_features(cxxeffect INTERFACE cxx_std_20)
+
+# boilerplate compiler options to have support for C++20. We can remove if you don't want them.
```

Comment:
Comment no longer needed here, I think?

Reply:
Yes. I'll add comments explaining the additional compiler options like "/EH" and remove that comment.